### PR TITLE
[SPARKR][SPARK-8452] expose jobGroup API in SparkR

### PR DESCRIPTION
**Pull request by falaki**
This pull request adds following methods to SparkR:

``` R
setJobGroup()
cancelJobGroup()
clearJobGroup()
```

For each method, the spark context is passed as the first argument. There does not seem to be a good way to test these in R.

cc @shivaram and @davies

### CODE REVIEW

File: R/pkg/R/sparkR.R [hunk 1]
```diff
@@ -278,3 +278,38 @@ sparkRHive.init <- function(jsc = NULL) {
...
+#' Assigns a group ID to all the jobs started by this thread until the group ID is set to a
+#' different value or cleared.
+#'
+#' @param sc The existing 
+#' @param groupid the ID to be assigned to job groups
+#' @param description description for the the job group ID
+#' @param interruptOnCancel flag to indicate if the job is interrupted on job cancellation
+
+setJobGroup <- function(groupId, description, interruptOnCancel) {
+  if (exists(".sparkRjsc", envir = env)) {
```
Comment (shivaram) [inline R/pkg/R/sparkR.R]: While its technically fine to just lookup the `SparkContext` I think all our methods take in a SQLContext / SparkContext explicitly. Will that work for your use case as well ?

Comment (falaki) [inline R/pkg/R/sparkR.R]: Yes, that is perfectly fine. Just wondering why doesn't sparkR.stop() follow that convention?

**Comment by davies** [comment]
LGTM, it will be better if you could add some tests for it, and add examples in the doc. Right now it's hard to tell what's the type of these parameters.

File: R/pkg/R/sparkR.R [hunk 2]
```diff
@@ -278,3 +278,32 @@ sparkRHive.init <- function(jsc = NULL) {
...
   hiveCtx
 }
+
+#' Assigns a group ID to all the jobs started by this thread until the group ID is set to a
+#' different value or cleared.
+#'
+#' @param sc existing spark context
+#' @param groupid the ID to be assigned to job groups
+#' @param description description for the the job group ID
+#' @param interruptOnCancel flag to indicate if the job is interrupted on job cancellation
```
Comment (shivaram) [inline R/pkg/R/sparkR.R]: Yeah @davies point is a good one. We can add an example usage here with something like 
```
 #' @examples
 #'\dontrun{
 #' sc <- sparkR.init()
 # setJobGroup(sc, "group", "some group", TRUE)
 #'}
```

**Comment by davies** [comment]
LGTM, waiting for tests.

**Comment by shivaram** [comment]
Thanks @falaki -- Merging this

**Pull request closed without merging**